spark/src/main/scala/org/apache/spark/sql/delta/TypeWidening.scala

```diff
@@ -62,6 +62,19 @@ object TypeWidening {
       case _ => false
     }
 
+  /**
+   * Returns whether the given type change can be applied during schema evolution. Only a
+   * subset of supported type changes are considered for schema evolution.
+   */
+  def isTypeChangeSupportedForSchemaEvolution(fromType: AtomicType, toType: AtomicType): Boolean =
+    (fromType, toType) match {
+      case (from, to) if !isTypeChangeSupported(from, to) => false
+      case (from, to) if from == to => true
+      case (ByteType, ShortType) => true
+      case (ByteType | ShortType, IntegerType) => true
```

Comment:
Nit: This can be replaced by
```scala
case (ByteType | ShortType, ShortType | IntegerType) => true`
```

Reply:
It's harder to read though, I'd rather keep this split